src/io/read_tree_lhalo_hdf5.c

```diff
@@ -237,51 +283,53 @@ int setup_forests_io_lht_hdf5(struct forest_info *forests_info, const int firstf
+        int status = read_dataset(fd, dataset_name, buffer);            \
         if (status != EXIT_SUCCESS) {                                   \
             ABORT(status);                                              \
         }                                                               \
         for (int halo_idx = 0; halo_idx < nhalos; ++halo_idx) {         \
             for (int dim = 0; dim < NDIM; ++dim) {                      \
-                local_halos[halo_idx].sage_name[dim] = ((C_dtype*)buffer_multipledim)[halo_idx * NDIM + dim]; \
+                local_halos[halo_idx].sage_name[dim] = ((C_dtype*) buffer)[halo_idx * NDIM + dim]; \
             }                                                           \
         }                                                               \
     }
 
 
-int64_t load_forest_hdf5(const int32_t forestnr, struct halo_data **halos, struct forest_info *forests_info)
+int64_t load_forest_hdf5(const int64_t forestnr, struct halo_data **halos, struct forest_info *forests_info, const double hubble)
 {
     char dataset_name[MAX_STRING_LEN];
-    int32_t status;
-
-    double *buffer; // Buffer to hold the read HDF5 data.
-    // The largest data-type will be double.
-
-    double *buffer_multipledim; // However also need a buffer three times as large to hold data such as position/velocity.
+    void *buffer; // Buffer to hold the read HDF5 data.
 
     /* const int64_t nhalos = (int64_t) forests_info->lht.nhalos_per_forest[forestnr];/\* the array itself contains int32_t, since the LHT format*\/ */
     hid_t fd = forests_info->lht.h5_fd[forestnr];
+
     /* CHECK: HDF5 file pointer is valid */
     if(fd <= 0 ) {
         fprintf(stderr,"Error: File pointer is NULL (i.e., you need to open the file before reading).\n"
                 "This error should already have been caught before reaching this line\n");
         ABORT(INVALID_FILE_POINTER);
     }
 
+    const int64_t treenum_in_file = forests_info->original_treenr[forestnr];
+    /* fprintf(stderr,"forestnr = %d fd = %d treenum = %d\n", forestnr, (int) fd, treenum_in_file); */
 
-    /* Figure out nhalos by checking the size of the 'Descendant' dataset */
+    
+    /* Figure out nhalos by checking the size of the 'Descendant' dataset (could be any other valid field) */
     /* https://stackoverflow.com/questions/15786626/get-the-dimensions-of-a-hdf5-dataset */
     const char field_name[] = "Descendant";
-    snprintf(dataset_name, MAX_STRING_LEN - 1, "tree_%03d/%s", forestnr, field_name);
+    snprintf(dataset_name, MAX_STRING_LEN - 1, "Tree%"PRId64"/%s", treenum_in_file, field_name);
     hid_t dataset = H5Dopen(fd, dataset_name, H5P_DEFAULT);
     hid_t dspace = H5Dget_space(dataset);
     const int ndims = H5Sget_simple_extent_ndims(dspace);
-    XASSERT(ndims == 0, -1,
-            "Error: Expected field = '%s' to be 1-D array\n",
-            field_name);
+    XASSERT(ndims == 1, -1,
```

Comment:
If this function handles it, `return EXIT_FAILURE` instead.

Reply:
Since these functions return an actual value, I have resorted to returning the negative of the error code to indicate failed status. The caller checks `if (status < 0)` and returns if that is the case